docs/conf.py

```
@@ -240,6 +251,32 @@ def _copy_md_sources(app, exception):
+    if exception is not None or app.builder.name not in (
+            'html', 'readthedocs'):
+        return
+    from pathlib import Path
+    from jinja2 import Environment, StrictUndefined
+    tpl_path = Path(app.srcdir) / '_templates' / 'llms.txt.j2'
+    out_path = Path(app.outdir) / 'llms.txt'
+    baseurl = (app.config.html_baseurl or '').rstrip('/') + '/'
+    # StrictUndefined: missing template variables raise rather than
+    # silently render as empty strings, so a typo in llms.txt.j2 fails
+    # the build instead of shipping a half-blank llms.txt.
+    env = Environment(undefined=StrictUndefined, keep_trailing_newline=True)
```

Ruff (0.15.12)

[error] By default, jinja2 sets `autoescape` to `False`. Consider using `autoescape=True` or the `select_autoescape` function to mitigate XSS vulnerabilities.

(S701)